cmd/deploy.go

```diff
@@ -173,7 +169,26 @@ func runDeploy(cmd *cobra.Command, _ []string, newClient ClientFactory) (err err
 			return err
 		}
 	case fn.BuildTypeGit:
-		return client.RunPipeline(cmd.Context(), config.Path)
+		git := function.Git
```

Comment:
Should we give any thought to the fact that a user could type `func deploy --git-url https://github.com...`. I realize it's a strange scenario and the user should not do it. But since `--build local` is the default, what does it mean if they specify a git URL. I think probably, the CLI should emit a warning message clarifying that you must specify `--build git` if any of these new flags are used, but `--build` is not.